src/rank_llm/retrieve/pyserini_retriever.py

```diff
@@ -169,28 +170,21 @@ def retrieve_and_store(
         Path(f"retrieve_results/{self._retrieval_method.name}").mkdir(
             parents=True, exist_ok=True
         )
+        writer = ResultsWriter(results)
         # Store JSON in rank_results to a file
-        with open(
-            f"retrieve_results/{self._retrieval_method.name}/retrieve_results_{self._dataset}.json",
-            "w",
-        ) as f:
-            json.dump(results, f, indent=2)
+        writer.write_in_json_format(
+            f"retrieve_results/{self._retrieval_method.name}/retrieve_results_{self._dataset}.json"
+        )
         # Store the QRELS of the dataset if specified
         if store_qrels:
             Path("qrels/").mkdir(parents=True, exist_ok=True)
             with open(f"qrels/qrels_{self._dataset}.json", "w") as f:
                 json.dump(self._qrels, f, indent=2)
         # Store TRECS if specified
         if store_trec:
-            with open(
-                f"retrieve_results/{self._retrieval_method.name}/trec_results_{self._dataset}.txt",
-                "w",
-            ) as f:
-                for result in results:
-                    for hit in result["hits"]:
-                        f.write(
-                            f"{hit['qid']} Q0 {hit['docid']} {hit['rank']} {hit['score']} rank\n"
-                        )
+            writer.write_in_trec_eval_format(
+                f"retrieve_results/{self._retrieval_method.name}/trec_results_{self._dataset}.txt"
```

Comment:
Can this naming be automated in writer too?

Reply:
I prefer to keep it this way since: 1- the writer is shared between retriever and reranker. 2- all this information about the dataset, ertrieval method, etc should be passed to the writer to be able generate the names properly.